src/Microsoft.AspNet.Mvc.Razor/HelperResult.cs

```diff
@@ -14,24 +15,26 @@ namespace Microsoft.AspNet.Mvc.Razor
     /// </summary>
     public class HelperResult : IHtmlContent
     {
-        private readonly Action<TextWriter> _action;
+        private readonly Func<TextWriter, Task> _asyncAction;
```

Comment:
Is this really the only option to fix this bug? Should whatever is producing a helper result today be returning a `Task<IHtmlContent>`?